content/topics/Analyze/causal-inference/did/honest-did.md

```diff
@@ -0,0 +1,317 @@
+
+{{% codeblock %}}
+```R
+
+# 3.1. Staggered DiD analysis 
+
+staggered_results <- did::att_gt(yname = "rating", # outcome var
+                          tname = "period_review", # time var
+                          idname = "asin", # individual var
+                          gname = "period_question", # treated
+                          control_group = "notyettreated", #control
+                          data = goodreads_df, # final dataset
+                          allow_unbalanced_panel = TRUE,
+                          base_period = "universal")
+
+# Aggregate results 
+es <- did::aggte(cs_results, type = "dynamic", na.rm = TRUE)
+
+
+# 3.2 SENSITIVITY ANALYSIS: RELATIVE MAGNITUDES 
+
+sensitivity_results_rm <-
+  honest_did(es,
+             e=0,
+             type="relative_magnitude",
+             Mbarvec=seq(from = 0.5, to = 2, by = 0.5))
+
+HonestDiD::createSensitivityPlot_relativeMagnitudes(sensitivity_results_rm$robust_ci,
+                                                    sensitivity_results_rm$orig_ci)
+
+
+# 3.3 SENSITIVITY ANALYSIS: SMOOTHNESS RESTRICTIONS
+
+sensitivity_results_sd <-
+  honest_did(es = es,
+             e = 0,
+             type="smoothness")
+
+
+HonestDiD::createSensitivityPlot(sensitivity_results_sd$robust_ci,
+                                sensitivity_results_sd$orig_ci)
+
+```
+{{% /codeblock %}}
+
+The following plots are generated:
+
+1. *Relative magnitudes*
+
+<p align = "center">
+<img src = "../images/staggered-rm-plot.png" width="500">
+</p>
+
+
+2. *Smoothness restriction*
+
+<p align = "center">
+<img src = "../images/staggered-sd-plot.png" width="500">
+</p>
+
+{{% summary %}}
+ 
+The `HonestDiD` approach by Rambachan & Roth (2019) addresses violations of the parallel trends assumption in DiD analysis by extrapolating pre-treatment differences to the post-treatment period and allowing for an interval of differences. This method can serve as a sensitivity analysis, enhancing the credibility of causal inference in DiD studies!
+
+While the assumptions about the relationship between post- and pre-trend differences are very context-specific, there are two main approaches to setting restrictions and specifying your framework (with possible combinations of these approaches):
+
+1. *Relative magnitude bounds*: Sets limits based on pre-treatment differences. 
+2. *Smoothness restriction*: Assumes trends evolve smoothly over time. 
+
+{{% /summary %}}
+
+
+### References 
+
+- [Rambachan and Ruth (2019) - A More Credible Approach to Parallel Trends](https://www.jonathandroth.com/assets/files/HonestParallelTrends_Main.pdf)
+
+- [GitHub Page HonestDiD, Example Application](https://github.com/asheshrambachan/HonestDiD)
+
+- [Blog post of David Mckenzie: What happens if the parallel trends assumption is (might be) violated?](https://blogs.worldbank.org/en/impactevaluations/revisiting-difference-differences-parallel-trends-assumption-part-ii-what-happens)
+
```

Comment:
Great work @valerievossen ! I think the overall writing and flow are good -- @srosh2000 could you double check the gists to see whether everything aligns with your previous BBs?